content/departments/engineering/dev/tools/grpc_cs_enablement.md

```diff
@@ -0,0 +1,110 @@
+- Well-defined semantics for backwards compatibility
+- Well-typed APIs with code generation across a wide variety of programming languages
+
+See the [PR-FAQ for more background](https://docs.google.com/document/d/1lf7REi_4t03tIxSPccDG4VHtNo-6SLI1-3zDcrspi6k/edit).
+
+## What do customers need to know?
+
+gRPC **\*\*\*\***\*\*\*\***\*\*\*\***only affects traffic **_between_** our microservices (e.x. `searcher` ↔ `gitserver`). Traffic between the Sourcegraph Web UI and the rest of the application is unaffected (e.x. `[sourcegraph.example.com](http://sourcegraph.example.com)` ↔ `frontend`’s GraphQL API).
+
+**If you don’t have any security restrictions on Sourcegraph’s _internal_ traffic, your customer shouldn’t need to tweak anything.**
+
+Otherwise, they might need to tweak their security settings / firewall configuration to allow this new type. The following is some general information about gRPC that can help customers change any relevant settings:
+
+- **Protocol Description**: gRPC runs on-top of [HTTP/2](https://en.wikipedia.org/wiki/HTTP/2). It transfers (binary-encoded, not human-readable plain-text) [Protocol Buffer](https://protobuf.dev/) payloads. Our current gRPC implementation does not use any encryption.
+- **List of services**: The following services will now _speak mainly gRPC in addition_ to their previous traffic:
+  - [frontend](https://github.com/sourcegraph/deploy-sourcegraph/blob/master/base/frontend/sourcegraph-frontend.Service.yaml)
+  - [gitserver](https://github.com/sourcegraph/deploy-sourcegraph-cloud/blob/release/base/gitserver/gitserver.Service.yaml)
+  - [zoekt-webserver](https://github.com/sourcegraph/deploy-sourcegraph-cloud/blob/release/base/indexed-search/indexed-search.StatefulSet.yaml)
+  - [zoekt-indexserver](https://github.com/sourcegraph/deploy-sourcegraph-cloud/blob/release/base/indexed-search/indexed-search.StatefulSet.yaml)
+  - [symbols](https://github.com/sourcegraph/deploy-sourcegraph-cloud/blob/release/base/symbols/symbols.Deployment.yaml)
+  - [repo-updater](https://github.com/sourcegraph/deploy-sourcegraph-cloud/blob/release/base/repo-updater/repo-updater.Deployment.yaml)
+- The following aspects about Sourcegraph’s networking configuration **aren’t changing**:
+  - \***\*\*\*\*\***Ports:\***\*\*\*\*\*** all Sourcegraph services will use the same ports as they were in the **5.1.X** release.
+  - **External traffic**: gRPC only affects how Sourcegraph’s microservices communicate amongst themselves - **no new external traffic is sent via gRPC**.
+  - **Service dependencies:** each Sourcegraph service will communicate with the same set of services regardless of whether gRPC is enabled.
+    - Example: `searcher` will still need to communicate with `gitserver` to fetch repository data. Whether or not gRPC is enabled doesn’t matter.
+
+## enabling / disabling GRPC
+
+gRPC will be the only means of communication used in Sourcegraph `5.3.X`. In the `5.2.x` release, you can disable the use of gRPC by:
```

Comment:
Can you explain whether it's enabled by default in 5.2? This reads to me like it can be disabled but doesn't explicitly say whether it is enabled by default. It'd make it explicit, maybe with a list?

```
- `5.2`: enabled by default, can be turned off via feature flag
- `5.3`: enabled by default, cannot be turned off

Reply:
I added a table to make this clear:

> ## Rollout timeline
> 
> | Sourcegraph version                   | gRPC feature status                                                                   |
> | ------------------------------------- | ------------------------------------------------------------------------------------- |
> | 5.2.X (releasing October 4th, 2023)   | On by default, can be disabled via feature flag (see “enabling/disabling GRPC below”) |
> | 5.3.X (releasing December 14th, 2023) | On by default, cannot be turned off                                                   |
>

Reply:
Hell yes